File: Java/T7/codigos2021/T07_ViewPager/app/src/main/java/com/example/t07_viewpager/FragmentUno.java
package com.example.t07_viewpager;

import android.content.Context;
import android.os.Bundle;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

public class FragmentUno extends Fragment {

    @Override
    public void onAttach(@NonNull Context context) {
        super.onAttach(context);
        Log.v("fragment","OnAttach 1");
    }

    public void cambiarTexto(){
        // cambia el texto de algo que hay dentro
    }

    @Override
    public void onDetach() {
        super.onDetach();
        Log.v("fragment","OnDettach 1");


    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.layout_fragment_uno, container,false);
        return view;
    }
}

File: Kotlin/T7/codigos2223/InicioFragments/app/src/main/java/com/example/iniciofragments/fragments/FragmentUno.kt
package com.example.iniciofragments.fragments

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import com.example.iniciofragments.R
import com.example.iniciofragments.databinding.FragmentHeaderBinding
import com.example.iniciofragments.databinding.FragmentUnoBinding
import com.google.android.material.snackbar.Snackbar

class FragmentUno: Fragment() {

    private lateinit var binding: FragmentUnoBinding

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        binding = FragmentUnoBinding.inflate(inflater,container,false)
        return binding.root
    }


    override fun onStart() {
        super.onStart()
        binding.botonNombreF1.setOnClickListener {
            //it -> el elemento que provoca el evento boton
            if (binding.editNombreF1.text.isEmpty()){
                Snackbar.make(binding.root, "No hay nombre a mostrar",Snackbar.LENGTH_SHORT).show()
            } else {
                Snackbar.make(binding.root, binding.editNombreF1.text.toString(),Snackbar.LENGTH_SHORT).show()
            }

        }
    }


}